web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/SelectSecurityGroups.tsx

```diff
@@ -187,3 +197,111 @@ export const SelectSecurityGroups = ({
     </>
   );
 };
+
+function withTips(
+  securityGroups: SecurityGroup[],
+  db: AwsRdsDatabase
+): SecurityGroupWithRecommendation[] {
+  const trustedGroups = getTrustedSecurityGroups(securityGroups, db);
+  return securityGroups.map(group => {
+    const isTrusted = trustedGroups.has(group.id);
+    const isOutboundAllowed = allowsOutbound(group);
+    return {
+      ...group,
+      tips: getTips(db, isTrusted, isOutboundAllowed),
+      // we recommend when either are true because security group rules are
+      // additive, meaning they can select multiple groups for a combined effect
+      // of satisfying the database inbound rules and the ECS task outbound
+      // rules.
+      recommended: isTrusted || isOutboundAllowed,
+    };
+  });
+}
+
+function getTips(
+  db: AwsRdsDatabase,
+  isTrusted: boolean,
+  allowsOutbound: boolean
+): string[] {
+  const result: string[] = [];
+  if (isTrusted) {
+    result.push(
+      `The database security ${pluralize(db.securityGroups.length, 'group')} allow inbound traffic from this security group`
+    );
+  }
+  if (allowsOutbound) {
+    result.push('This security group allows outbound traffic to the internet');
+  }
+  return result;
+}
+
+function allowsOutbound(sg: SecurityGroup): boolean {
+  return sg.outboundRules.some(rule => {
+    const havePorts = allowsOutboundToPorts(rule);
+    // this is a heuristic, because an exhaustive analysis is non-trivial.
+    return havePorts && rule.cidrs.some(cidr => cidr.cidr === '0.0.0.0/0');
+  });
+}
+
+function allowsOutboundToPorts(rule: SecurityGroupRule): boolean {
+  const publicECRPort = 443;
+  const proxyPort = window.location.port;
+  if (!proxyPort || proxyPort == '443') {
+    // if proxy port is not found or it is also 443, then we only check for
+    // the HTTPS port.
+    return ruleAllowsPort(rule, publicECRPort);
+  }
+  // otherwise we need to check that the rule allows both proxy and ECR ports.
+  return (
+    ruleAllowsPort(rule, publicECRPort) &&
+    ruleAllowsPort(rule, parseInt(proxyPort, 10))
+  );
+}
+
+function getTrustedSecurityGroups(
+  securityGroups: SecurityGroup[],
+  db: AwsRdsDatabase
+): Set<string> {
+  const securityGroupsById = new Map<string, SecurityGroup>();
+  securityGroups.forEach(group => {
+    securityGroupsById.set(group.id, group);
+  });
```

Comment:
```suggestion
  const securityGroupsById = new Map(
    securityGroups.map(group => [group.id, group])
  );
```

Reply:
good to know this is possible (you can probably tell i'm a js noob 😉 )